muxers/yamux/src/lib.rs

```diff
@@ -39,21 +38,19 @@ use thiserror::Error;
 use yamux::ConnectionError;
 
 /// A Yamux connection.
-pub struct Yamux<S>(Mutex<Inner<S>>);
+pub struct Yamux<S> {
+    /// The [`futures::stream::Stream`] of incoming substreams.
+    incoming: S,
+    /// Handle to control the connection.
+    control: yamux::Control,
+}
 
 impl<S> fmt::Debug for Yamux<S> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.write_str("Yamux")
     }
 }
 
-struct Inner<S> {
```

Comment:
:tada:

Reply:
To expand on this: I am very happy for the `Mutex` around `Inner` to go away. Thanks @thomaseizinger!

Reply:
Me too 😊